src/js/client/ajaxCapable.js

```diff
@@ -62,18 +62,40 @@ A few more things to note:
         myTransaction.commit();
     };
 
+    // Strip null values to avoid validation errors with empty non-required fields.
+    // TODO:  Figure out why there are so many null values to strip
+    // TODO:  If this is required, add tests for it.
+    gpii.templates.ajaxCapable.stripNullValues = function (mapToStrip) {
```

Comment:
I think a function this bulky really does require a test, yes - can you give an example of a layout in which the null values occur?